haskell-language-server.cabal

```diff
@@ -205,69 +205,59 @@ executable haskell-language-server-wrapper
     , process
   default-language:    Haskell2010
 
-
 test-suite func-test
-  import: agpl
-  type: exitcode-stdio-1.0
-  default-language: Haskell2010
-  build-tool-depends:  hspec-discover:hspec-discover
-                     , haskell-language-server:haskell-language-server
-                     , cabal-helper:cabal-helper-main
-                     , ghcide:ghcide-test-preprocessor
-
-  build-depends:
-      base >=4.7 && <5
-    , aeson
-    , data-default
-    , haskell-lsp-types == 0.22.*
-    , hls-test-utils
-    , hspec
-    , lens
-    , lsp-test >= 0.10.0.0
-    , text
-    , unordered-containers
-  other-modules:
-                     --  CompletionSpec
-                     -- , CommandSpec
-                     -- , DeferredSpec
-                     -- , DefinitionSpec
-                     -- , DiagnosticsSpec
-                      FormatSpec
-                     -- , FunctionalBadProjectSpec
-                     -- , FunctionalCodeActionsSpec
-                     -- , FunctionalLiquidSpec
-                     , FunctionalSpec
-                     -- , HaReSpec
-                     -- , HieBiosSpec
-                     -- , HighlightSpec
-                     -- , HoverSpec
-                     , PluginSpec
-                     -- , ProgressSpec
-                     -- , ReferencesSpec
-                     -- , RenameSpec
-                     -- , SymbolsSpec
-                     -- , TypeDefinitionSpec
-                     , Utils
-                     , Paths_haskell_language_server
-
-  hs-source-dirs:
-      test/functional
-  ghc-options:
-         -Wall
-         -Wredundant-constraints
-         -Wno-name-shadowing
-         -threaded -rtsopts -with-rtsopts=-N
+  import:               agpl
+  type:                 exitcode-stdio-1.0
+  default-language:     Haskell2010
+  build-tool-depends:   haskell-language-server:haskell-language-server
+                      , cabal-helper:cabal-helper-main
```

Comment:
The whole line is in my point of view unneeded. 
cc @alanz or did you add it for a reason?

Reply:
Well, we are running functional tests that actually run the server. If the test does not depend on the server, we are likely to end up running a functional tests against a stale build of the server.

I did not add it, but I think that is why.

Reply:
But this `cabal-helper:cabal-helper-main` should be unnecassary, right?

Reply:
Looks to me light an artefact from pre cabal-helper-1.0

Reply:
If we are not using cabal-helper for running the server, then it can be removed

Reply:
done